runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp

```diff
@@ -148,7 +146,9 @@ initImpl(J9VMThread *currentThread, j9object_t membernameObject, j9object_t refO
 	if (!VM_VMHelpers::exceptionPending(currentThread)) {
 		J9VMJAVALANGINVOKEMEMBERNAME_SET_FLAGS(currentThread, membernameObject, flags);
 		J9VMJAVALANGINVOKEMEMBERNAME_SET_NAME(currentThread, membernameObject, nameObject);
-		J9VMJAVALANGINVOKEMEMBERNAME_SET_TYPE(currentThread, membernameObject, typeObject);
+		if (NULL != typeObject) {
```

Comment:
Can you add an VM Assert here to validate this is only done for fields?  something like `Trc_VM_Assert(refClass == J9VMJAVALANGREFLECTFIELD(vm))`

Reply:
Added: `Assert_JCL_true(OMR_ARE_ALL_BITS_SET(flags, MN_IS_FIELD))`.